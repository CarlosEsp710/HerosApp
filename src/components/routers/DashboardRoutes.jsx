import React from "react";
import { Routes, Route } from "react-router-dom";

import { DCScreen } from "../DC/DCScreen";
import { HeroScreen } from "../hero/HeroScreen";
import { MarvelScreen } from "../Marvel/MarvelScreen";
import { SearchScreen } from "../search/SearchScreen";
import { NavBar } from "../ui/NavBar";

export const DashboardRoutes = () => {
  return (
    <>
      <NavBar></NavBar>
      <Routes>
        <Route path="marvel" element={<MarvelScreen />} />
        <Route path="dc" element={<DCScreen />} />
        <Route path="search" element={<SearchScreen />} />
        <Route path="hero/:heroId" element={<HeroScreen />} />
        <Route path="/" element={<MarvelScreen />} />
      </Routes>
    </>
  );
};
