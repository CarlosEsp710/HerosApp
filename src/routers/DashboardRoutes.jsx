import React from "react";
import { Routes, Route } from "react-router-dom";

import { DCScreen } from "../components/DC/DCScreen";
import { HeroScreen } from "../components/hero/HeroScreen";
import { MarvelScreen } from "../components/Marvel/MarvelScreen";
import { SearchScreen } from "../components/search/SearchScreen";
import { NavBar } from "../components/ui/NavBar";

export const DashboardRoutes = () => {
  return (
    <>
      <NavBar></NavBar>
      <Routes>
        <Route path="marvel" element={<MarvelScreen />} />
        <Route path="dc" element={<DCScreen />} />
        <Route path="search" element={<SearchScreen />} />
        <Route path="hero/:heroId" element={<HeroScreen />} />
        <Route path="/" element={<MarvelScreen />} />
      </Routes>
    </>
  );
};
