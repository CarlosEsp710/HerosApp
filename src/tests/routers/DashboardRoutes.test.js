import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import { AuthContext } from "../../auth/authContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";

describe("DashboardRoutes", () => {
  const authContext = {
    user: {
      isLoggedIn: true,
      name: "Carlos Espejel",
      email: "carlosespejel71@gmail.com",
      imageUrl: "https://avatars.githubusercontent.com/u/57500554?s=96&v=4",
    },
  };

  test("should return properly - MarvelScreen", () => {
    const wrapper = mount(
      <AuthContext.Provider value={authContext}>
        <MemoryRouter initialEntries={["/"]}>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h1").text()).toBe("Marvel Screen");
  });

  test("should return properly - DCScreen", () => {
    const wrapper = mount(
      <AuthContext.Provider value={authContext}>
        <MemoryRouter initialEntries={["/dc"]}>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h1").text()).toBe("DC Screen");
  });

  test("should return properly - SearchScreen", () => {
    const wrapper = mount(
      <AuthContext.Provider value={authContext}>
        <MemoryRouter initialEntries={["/search"]}>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("h1").text()).toBe("Search Screen");
  });
});
