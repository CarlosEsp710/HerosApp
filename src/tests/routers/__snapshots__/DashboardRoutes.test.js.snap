// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRoutes should return properly - DCScreen 1`] = `
Array [
  <div
    className="min-h-full"
  >
    <nav
      className="bg-gray-800"
    >
      <div
        className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
      >
        <div
          className="flex items-center justify-between h-16"
        >
          <div
            className="flex items-center"
          >
            <div
              className="flex-shrink-0"
            >
              <img
                alt="Workflow"
                className="h-8 w-8"
                src="https://tailwindui.com/img/logos/workflow-mark-indigo-500.svg"
              />
            </div>
            <div
              className="hidden md:block"
            >
              <div
                className="ml-10 flex items-baseline space-x-4"
              >
                <a
                  aria-current="page"
                  className="bg-gray-900 text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/"
                  onClick={[Function]}
                >
                  Dashboard
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/marvel"
                  onClick={[Function]}
                >
                  Marvel
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/dc"
                  onClick={[Function]}
                >
                  DC
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/search"
                  onClick={[Function]}
                >
                  Search
                </a>
                <button
                  className="hover:bg-gray-100 block px-4 py-2 text-sm text-gray-700"
                  onClick={[Function]}
                >
                  Logout
                </button>
              </div>
            </div>
            <div
              className="hidden md:block justify-end"
            >
              <div
                className="ml-4 flex items-center md:ml-6"
              >
                <div
                  className="ml-3 relative"
                >
                  <div>
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="max-w-xs bg-gray-800 rounded-full flex items-center text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                      id="headlessui-menu-button-6"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <span
                        className="sr-only"
                      >
                        Open user menu
                      </span>
                      <img
                        alt="Carlos Espejel"
                        className="h-8 w-8 rounded-full"
                        src="https://avatars.githubusercontent.com/u/57500554?s=96&v=4"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="-mr-2 flex md:hidden"
          >
            <button
              aria-expanded={false}
              className="bg-gray-800 inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
              id="headlessui-disclosure-button-4"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                className="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                className="block h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  Array [
    <header
      className="bg-white shadow"
    >
      <div
        className="max-w-7xl mx-auto py-6 px-4 sm:px-6 lg:px-8"
      >
        <h1
          className="text-3xl font-bold text-gray-900"
        >
          DC Screen
        </h1>
      </div>
    </header>,
    <div
      className="container"
    >
      <div
        className="grid grid-cols-1 gap-y-10 sm:grid-cols-2 gap-x-6 lg:grid-cols-3 xl:grid-cols-4 xl:gap-x-8"
      >
        <a
          className="group"
          href="/hero/dc-batman"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Batman"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-batman.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Bruce Wayne
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Batman
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-superman"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Superman"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-superman.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Kal-El
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Superman
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-flash"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Flash"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-flash.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Jay Garrick
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Flash
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-green"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Green Lantern"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-green.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Alan Scott
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Green Lantern
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-arrow"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Green Arrow"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-arrow.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Oliver Queen
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Green Arrow
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-wonder"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Wonder Woman"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-wonder.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Princess Diana
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Wonder Woman
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-martian"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Martian Manhunter"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-martian.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            J'onn J'onzz
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Martian Manhunter
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-robin"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Robin/Nightwing"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-robin.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Dick Grayson
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Robin/Nightwing
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-blue"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Blue Beetle"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-blue.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Dan Garret
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Blue Beetle
          </p>
        </a>
        <a
          className="group"
          href="/hero/dc-black"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Black Canary"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/dc-black.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Dinah Drake
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Black Canary
          </p>
        </a>
      </div>
    </div>,
  ],
]
`;

exports[`DashboardRoutes should return properly - MarvelScreen 1`] = `
Array [
  <div
    className="min-h-full"
  >
    <nav
      className="bg-gray-800"
    >
      <div
        className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
      >
        <div
          className="flex items-center justify-between h-16"
        >
          <div
            className="flex items-center"
          >
            <div
              className="flex-shrink-0"
            >
              <img
                alt="Workflow"
                className="h-8 w-8"
                src="https://tailwindui.com/img/logos/workflow-mark-indigo-500.svg"
              />
            </div>
            <div
              className="hidden md:block"
            >
              <div
                className="ml-10 flex items-baseline space-x-4"
              >
                <a
                  aria-current="page"
                  className="bg-gray-900 text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/"
                  onClick={[Function]}
                >
                  Dashboard
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/marvel"
                  onClick={[Function]}
                >
                  Marvel
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/dc"
                  onClick={[Function]}
                >
                  DC
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/search"
                  onClick={[Function]}
                >
                  Search
                </a>
                <button
                  className="hover:bg-gray-100 block px-4 py-2 text-sm text-gray-700"
                  onClick={[Function]}
                >
                  Logout
                </button>
              </div>
            </div>
            <div
              className="hidden md:block justify-end"
            >
              <div
                className="ml-4 flex items-center md:ml-6"
              >
                <div
                  className="ml-3 relative"
                >
                  <div>
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="max-w-xs bg-gray-800 rounded-full flex items-center text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                      id="headlessui-menu-button-1"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <span
                        className="sr-only"
                      >
                        Open user menu
                      </span>
                      <img
                        alt="Carlos Espejel"
                        className="h-8 w-8 rounded-full"
                        src="https://avatars.githubusercontent.com/u/57500554?s=96&v=4"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="-mr-2 flex md:hidden"
          >
            <button
              aria-expanded={false}
              className="bg-gray-800 inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
              id="headlessui-disclosure-button-2"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                className="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                className="block h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  Array [
    <header
      className="bg-white shadow"
    >
      <div
        className="max-w-7xl mx-auto py-6 px-4 sm:px-6 lg:px-8"
      >
        <h1
          className="text-3xl font-bold text-gray-900"
        >
          Marvel Screen
        </h1>
      </div>
    </header>,
    <div
      className="container"
    >
      <div
        className="grid grid-cols-1 gap-y-10 sm:grid-cols-2 gap-x-6 lg:grid-cols-3 xl:grid-cols-4 xl:gap-x-8"
      >
        <a
          className="group"
          href="/hero/marvel-spider"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Spider Man"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-spider.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Peter Parker
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Spider Man
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-captain"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Captain America"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-captain.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Steve Rogers
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Captain America
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-iron"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Iron Man"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-iron.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Tony Stark
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Iron Man
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-thor"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Thor"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-thor.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Thor Odinson
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Thor
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-hulk"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Hulk"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-hulk.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Bruce Banner
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Hulk
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-wolverine"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Wolverine"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-wolverine.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            James Howlett
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Wolverine
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-daredevil"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Daredevil"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-daredevil.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Matthew Michael Murdock
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Daredevil
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-hawkeye"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Hawkeye"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-hawkeye.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Clinton Francis Barton
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Hawkeye
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-cyclops"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Cyclops"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-cyclops.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Scott Summers
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Cyclops
          </p>
        </a>
        <a
          className="group"
          href="/hero/marvel-silver"
          onClick={[Function]}
        >
          <div
            className="w-full aspect-w-1 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-8"
          >
            <img
              alt="Silver Surfer"
              className="w-full h-full object-center object-cover group-hover:opacity-75"
              src="/assets/img/marvel-silver.jpg"
            />
          </div>
          <h3
            className="mt-4 text-sm text-gray-700"
          >
            Norrin Radd
          </h3>
          <p
            className="mt-1 text-lg font-medium text-gray-900"
          >
            Silver Surfer
          </p>
        </a>
      </div>
    </div>,
  ],
]
`;

exports[`DashboardRoutes should return properly - SearchScreen 1`] = `
Array [
  <div
    className="min-h-full"
  >
    <nav
      className="bg-gray-800"
    >
      <div
        className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
      >
        <div
          className="flex items-center justify-between h-16"
        >
          <div
            className="flex items-center"
          >
            <div
              className="flex-shrink-0"
            >
              <img
                alt="Workflow"
                className="h-8 w-8"
                src="https://tailwindui.com/img/logos/workflow-mark-indigo-500.svg"
              />
            </div>
            <div
              className="hidden md:block"
            >
              <div
                className="ml-10 flex items-baseline space-x-4"
              >
                <a
                  aria-current="page"
                  className="bg-gray-900 text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/"
                  onClick={[Function]}
                >
                  Dashboard
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/marvel"
                  onClick={[Function]}
                >
                  Marvel
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/dc"
                  onClick={[Function]}
                >
                  DC
                </a>
                <a
                  className="text-gray-300 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="/search"
                  onClick={[Function]}
                >
                  Search
                </a>
                <button
                  className="hover:bg-gray-100 block px-4 py-2 text-sm text-gray-700"
                  onClick={[Function]}
                >
                  Logout
                </button>
              </div>
            </div>
            <div
              className="hidden md:block justify-end"
            >
              <div
                className="ml-4 flex items-center md:ml-6"
              >
                <div
                  className="ml-3 relative"
                >
                  <div>
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="max-w-xs bg-gray-800 rounded-full flex items-center text-sm focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                      id="headlessui-menu-button-9"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      type="button"
                    >
                      <span
                        className="sr-only"
                      >
                        Open user menu
                      </span>
                      <img
                        alt="Carlos Espejel"
                        className="h-8 w-8 rounded-full"
                        src="https://avatars.githubusercontent.com/u/57500554?s=96&v=4"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="-mr-2 flex md:hidden"
          >
            <button
              aria-expanded={false}
              className="bg-gray-800 inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
              id="headlessui-disclosure-button-7"
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              type="button"
            >
              <span
                className="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                className="block h-6 w-6"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h16M4 18h16"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  Array [
    <header
      className="bg-white shadow"
    >
      <div
        className="max-w-7xl mx-auto py-6 px-4 sm:px-6 lg:px-8"
      >
        <h1
          className="text-3xl font-bold text-gray-900"
        >
          Search Screen
        </h1>
      </div>
    </header>,
    <div
      className="container grid grid-cols-2"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="mt-1 block w-full px-3 py-2 bg-white border border-slate-300 rounded-md text-sm shadow-sm placeholder-slate-400
 focus:outline-none focus:border-sky-500 focus:ring-1 focus:ring-sky-500"
          complete="off"
          name="searchText"
          onChange={[Function]}
          placeholder="Search..."
          type="text"
          value=""
        />
        <button
          className="mt-5 w-full bg-transparent hover:bg-sky-300 text-sky-700 font-semibold hover:text-white py-2 px-4 border border-sky-500 hover:border-transparent rounded"
          type="submit"
        >
          Search
        </button>
      </form>
      <div
        className="ml-4"
      >
        <h4
          className="text-xl mb-4"
        >
          Results
        </h4>
        <div
          class="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative"
          role="alert"
        >
          <strong
            class="font-bold"
          >
            Holy smokes!
          </strong>
          <span
            class="block sm:inline"
          >
             Search a hero.
          </span>
        </div>
      </div>
    </div>,
  ],
]
`;
